jwst/extract_2d/grisms.py

```diff
@@ -192,7 +194,8 @@ def extract_tso_object(input_model,
             output_model.dq = ext_dq
             output_model.meta.wcs = subwcs
             output_model.meta.wcs.bounding_box = util.wcs_bbox_from_shape(ext_data.shape)
-            output_model.meta.wcs.crpix2 = 34  # update for the move, vals are the same
+            output_model.meta.wcsinfo.siaf_yref_sci = 34  # update for the move, vals are the same
+            output_model.meta.wcsinfo.siaf_xref_sci = input_model.meta.wcsinfo.siaf_xref_sci
```

Comment:
Not sure I understand the hard-coding of one and not the other here.  Also the hard-coding is in zero-indexed, not 1-indexed coordinates.  This keyword is supposed to be FITS, i.e. 1-indexed?

Reply:
The hard-coding is explained in the comments. It looks like it is supposed to be what it is for reasons not related to indexing.
Note, that the goal is to reproduce the previous code here, not invent new algorithms.

Reply:
The NIRCam TS folks wanted the extraction box moved a couple rows for the case where one of the subarrays is in use, such that the cutout is intentionally off-center a bit from the source location. That's why the hardcoded value for yref.